sources/packages/backend/workflow/test/2023-2024/parttime-assessment/appeals/parttime-assessment-student-dependants.e2e-spec.ts

```diff
@@ -0,0 +1,112 @@
+import { PROGRAM_YEAR } from "../../constants/program-year.constants";
+import {
+  ZeebeMockedClient,
+  createFakeConsolidatedPartTimeData,
+  executePartTimeAssessmentForProgramYear,
+} from "../../../test-utils";
+import {
+  createFakeStudentDependentEligible,
+  DependentEligibility,
+} from "../../../test-utils/factories";
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-student-dependants.`, () => {
+  it(
+    "Should have calculated student dependants variables assigned with request a change values " +
+      "when there is a request a change for student dependants.",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.appealsStudentDependantsData = [
+        createFakeStudentDependentEligible(
+          DependentEligibility.Eligible0To18YearsOld,
+          { referenceDate: assessmentConsolidatedData.offeringStudyStartDate },
+        ),
+        createFakeStudentDependentEligible(
+          DependentEligibility.Eligible18To22YearsOldAttendingHighSchool,
+          { referenceDate: assessmentConsolidatedData.offeringStudyStartDate },
+        ),
+        createFakeStudentDependentEligible(
+          DependentEligibility.Eligible18To22YearsOldDeclaredOnTaxes,
+          { referenceDate: assessmentConsolidatedData.offeringStudyStartDate },
+        ),
+        createFakeStudentDependentEligible(
+          DependentEligibility.EligibleOver22YearsOld,
+          { referenceDate: assessmentConsolidatedData.offeringStudyStartDate },
+        ),
+      ];
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+      // Assert
+      expect(
+        calculatedAssessment.variables.calculatedDataTotalEligibleDependants,
+      ).toBe(4);
+      expect(
+        calculatedAssessment.variables.calculatedDataDependants11YearsOrUnder,
+      ).toBe(0);
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataDependants12YearsOverOnTaxes,
+      ).toBe(2);
+      expect(
+        calculatedAssessment.variables
+          .calculatedDataTotalEligibleDependentsForChildCare,
+      ).toBe(2);
+    },
+  );
+
+  it(
```

Comment:
I would suggest to have a test with a dependent in student application but no dependant in student appeal.

Reply:
Test for an appeal with `no` for hasDependents is missing. 

![image](https://github.com/user-attachments/assets/645ea215-d54e-4886-b590-c47bbfd3b35e)